docs/source/dev/start.md

```diff
@@ -0,0 +1,130 @@
+to manage and install dependencies. Please note we require using `poetry>=1.2`.
+
+If you use `conda`, the above can be achieved with the following commands:
+
+```shell
+conda create --name hermes python=3.10 poetry
+```
+
+You should confirm whether to proceed.
+
+## Get the source code
+
+Next, you need to obtain a version of the HERMES source code.
+
+You can either download it as a zipped package or clone the whole Git repository.
+You can clone the repository and enter the project directory as follows:
+
+```shell
+git clone https://gitlab.com/hermes-hmc/workflow.git
+cd workflow
+```
+
+## Learn how our repo is structured
+
+- All Python sources are in the `src` folder
+- `pytest` tests are in the `test` folder
+- [Architectural Design Records (ADR)](https://adr.github.io/) are in `docs/adr`
+- API documentation is automatically generated into `docs/source/api`
+- All other Sphinx-based documentation lives in `docs/source/*`
+
+This project uses 
+
+- a development branch (`develop`) to merge developments into, this is the default branch
+- actual development is done on "feature" branches (this includes non-feature work such as bug fixing), see also our
+  {doc}`contribute`.
+- a `main` branch which only includes releases
+
+## Install HERMES and dependencies
+
+To finally install all the required dependencies and HERMES itself, you can now use `poetry`:
+
+```
+poetry install
+```
+
+### Which dependencies do we use?
+
+Our feature implementations build upon these main dependencies:
+
+- [`click`](https://click.palletsprojects.com/) for providing a commmand line interface
+- [`jsonschema`](https://python-jsonschema.readthedocs.io) to validate different JSON based file formats
+- [`ruaml.yaml`](https://yaml.readthedocs.io) to parse and validate YAML files
+- [`cffconvert`](https://github.com/citation-file-format/cff-converter-python) to harvest data from the CITATION.cff file
+
+Finally, there are some dependencies that are only required for active development. These are:
+
+- `pytest`, `pytest-cov` to run the developer tests and monitor the test coverage
+- `Sphinx`, `myst-parser` to build the documentation
+- `flake8` to check the coding style
+
+While there is no containerized or installable version of Hermes available yet, you need to install it manually into
+your Python environment for development, see below.
+
+
+## Verify installation works
+
+That's it, you should now have a working development copy of HERMES in your environment.
+You can confirm this by running `hermes`, seeing how it harvests all the information from the input:
+
+```shell
+hermes
+```
+
+## Verify tests can be run
+
+Tests are implemented using [pytest](https://pytest.org). You can generate coverage report using the `pytest-cov` plugin.
+Both tools are specified as development dependencies in the `pyproject.toml`.
+
+To verify tests with an extensive HTML report run, execute:
```

Comment:
```suggestion
To run all tests, execute `pytest test/` within the activated `poetry` environment.

To create an extensive test coverage report in HTML, execute:
```

Add the simple test run case as well, and change some wording.

Reply:
Good point!
Yet, I'd rather see it like this:

```suggestion
To run all tests within your activated environment, execute:

```shell
pytest test
```

To create an extensive test coverage report in HTML, to execute within a `poetry` managed environment, run:
```

Reply:
Depends on the result of the discussion above about introducing a world of different envs.